plugins/scaffolder/src/components/fields/Secret/Secret.test.tsx

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import React from 'react';
+import {
+  SecretsContextProvider,
+  useTemplateSecrets,
+} from '@backstage/plugin-scaffolder-react';
+import { Secret } from './Secret';
+import { renderInTestApp } from '@backstage/test-utils';
+import { Form } from '@backstage/plugin-scaffolder-react/alpha';
+import validator from '@rjsf/validator-ajv8';
+import { fireEvent, act } from '@testing-library/react';
+
+describe('<Secret/>', () => {
+  const SecretsComponent = () => {
+    const { secrets } = useTemplateSecrets();
+    return (
+      <div data-testid="current-secrets">{JSON.stringify({ secrets })}</div>
+    );
+  };
+
+  it('should set the current form value as a mask for the value entered', async () => {
+    const mockSecret = 'backstage';
+    const onSubmit = jest.fn();
+
+    const { getByLabelText, getByRole } = await renderInTestApp(
+      <SecretsContextProvider>
+        <Form
+          validator={validator}
+          schema={{ type: 'string', title: 'secret' }}
+          uiSchema={{
+            'ui:field': 'Secret',
+          }}
+          fields={{
+            Secret: Secret,
+          }}
+          onSubmit={onSubmit}
+        />
+        <SecretsComponent />
+      </SecretsContextProvider>,
+    );
+
+    const secretInput = getByLabelText('secret');
+    const submitButton = getByRole('button');
+
+    await act(async () => {
+      fireEvent.change(secretInput, { target: { value: mockSecret } });
```

Comment:
matter of taste: we should be using userEvent more eh, it's nicely compounding things and deals with `act` under the hood etc